### DISCUSSION TITLE
why dividing integers result becomes float ?

### DISCUSSION THREAD

**Discussion by shakibamoshiri** [mikefarah/yq#1687] [Q&A] [answered]
is this result valid, should not be integer ?
```bash
yq -n '.result=(200 / 10)'
result: !!float 20
```
I was expecting 
```bash
yq -n '.result=(200 / 10)'
result: 20
```

version 
```bash
yq --version 
yq (https://github.com/mikefarah/yq/) version v4.34.1
```

**Comment by mikefarah** [comment, marked as answer]
Yeah I think you're right - the divide is a little naive and just assumes a float. Feel free to raise a bug.

As a workaround you can manually override the tag to be an int:

```
./yq -n '.result = (20 / 10) | .result tag= "!!int"'
result: 2
```

**Reply by neiser** [reply]
@mikefarah Maybe you want to open this issue (see also my recent comment): https://github.com/mikefarah/yq/issues/69#issuecomment-2636522417
